.circleci/config.yml

```diff
@@ -1,5 +1,181 @@
+        macos:
+          xcode: 13.3.1
+        resource_class: large
+
+      android-executor:
+        machine:
+          image: android:202102-01
+        resource_class: large
+
+    commands:
+      run-with-retry:
+        description: Run command with retry
+        parameters:
+          label:
+            description: Display name
+            type: string
+          command:
+            description: Command to run
+            type: string
+          retry-count:
+            description: Number of retry
+            type: integer
+            default: 3
+          sleep:
+            description: Wait duration until next retry
+            type: integer
+            default: 5
+          no_output_timeout:
+            description: Elapsed time the command can run without output
+            type: string
+            default: 10m
+        steps:
+          - run:
+              name: << parameters.label >>
+              command: |
+                retry() {
+                  MAX_RETRY=<< parameters.retry-count >>
+                  n=0
+                  until [ $n -ge $MAX_RETRY ]
+                  do
+                      << parameters.command >> && break
+                      n=$[$n+1]
+                      sleep << parameters.sleep >>
+                  done
+                  if [ $n -ge $MAX_RETRY ]; then
+                    echo "failed: ${@}" >&2
+                    exit 1
+                  fi
+                }
+                retry
+              no_output_timeout: << parameters.no_output_timeout >>
+      install-flutter:
+        description: Install Flutter and set up paths.
+        parameters:
+          flutter_branch:
+            description: Flutter branch or version tag.
+            type: string
+            default: stable
+        steps:
+          - run:
+              name: Set up Flutter
+              command: |
+                echo 'export FLUTTER_HOME=${HOME}/sdks/flutter' >> $BASH_ENV
+                echo 'export FLUTTER_BRANCH=<< parameters.flutter_branch >>' >> $BASH_ENV
+                echo 'export FLUTTER_ROOT=${FLUTTER_HOME}' >> $BASH_ENV
+                echo 'export PATH=${PATH}:${FLUTTER_HOME}/bin:${FLUTTER_HOME}/bin/cache/dart-sdk/bin:${HOME}/.pub-cache/bin:${FLUTTER_HOME}/.pub-cache/bin' >> $BASH_ENV
+                source $BASH_ENV
+                git clone --branch ${FLUTTER_BRANCH} https://github.com/flutter/flutter.git ${FLUTTER_HOME}
+                (yes || true) | flutter doctor --android-licenses && flutter doctor
+                flutter precache
+      setup-amplify-flutter-project:
+        description: Setup Amplify project
+        steps:
+          - run-with-retry:
+              label: Setting up dependences
+              command: flutter pub add amplify_flutter && flutter pub add amplify_datastore && flutter pub add amplify_storage_s3 && flutter pub add amplify_analytics_pinpoint && flutter pub add amplify_auth_cognito
+              no_output_timeout: 20m
```

Comment:
This is a really long window. I would think `5m` is more than plenty. The problem is `pub` will retry indefinitely in certain situations so this will just burn compute time.

Reply:
oh I see, didn't know pub command will hang for some cases, will change it to 5m